gr-blocks/lib/file_source_impl.cc

```diff
@@ -188,7 +188,34 @@ void file_source_impl::open(const char* filename,
 
     // Rewind to start offset
     if (d_seekable) {
-        if (GR_FSEEK(d_new_fp, start_offset_items * d_itemsize, SEEK_SET) == -1) {
+        auto start_offset = start_offset_items * d_itemsize;
+#ifdef _POSIX_C_SOURCE
+#if _POSIX_C_SOURCE >= 200112L
+        // If supported, tell the OS that we'll be accessing the file sequentially
+        // and that it would be a good idea to start prefetching it
+        auto fd = fileno(d_new_fp);
+        static const std::map<int, const std::string> fadv_errstrings = {
```

Comment:
Just wondering: is there a reason for using this ad-hoc map instead of `strerror_r()`?

Reply:
not really, no.